### PR TITLE
Dropped / moved some of old testsuites

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

File: test/Makefile.am [hunk 1]
```diff
@@ -28,6 +28,7 @@ TESTS = \
   suse_firewall_4_network_test.rb \
   udev_test.rb \
   widgets_test.rb \
+  host_test.rb \
```
Comment (mvidner) [inline test/Makefile.am]: NP: keep the list in alphabetical order

File: src/modules/Host.rb [hunk 2]
```diff
@@ -207,14 +207,13 @@ def Import(settings)
...
     def Export
-      if Ops.greater_than(Builtins.size(@hosts), 0)
-        # Filter out IPs with empty hostname (so that valid autoyast
-        # profile is created)(#335120)
-        @hosts = Builtins.filter(@hosts) { |_ip, names| names != [] }
-        return { "hosts" => Builtins.eval(@hosts) }
-      else
-        return {}
-      end
+      return {} if hosts.empty?
```
Comment (mvidner) [inline src/modules/Host.rb]: Please add a test case for this too.

File: src/modules/Host.rb [hunk 3]
```diff
@@ -207,14 +207,13 @@ def Import(settings)
...
+
+      # Filter out IPs with empty hostname (so that valid autoyast
+      # profile is created)(#335120)
+      hosts.keep_if { |_, names| !names.empty? }
```
Comment (mvidner) [inline src/modules/Host.rb]:
```rb
    it "removes empty name lists" do
      allow(host).to receive(:hosts).and_return({"127.0.0.1" => ["localhost"], "10.0.0.1"  => []})
      expect(host.Export).to eql("hosts" => {"127.0.0.1" => ["localhost"]})
    end
```

Comment (jreidinger) [inline src/modules/Host.rb]: this creates quite nasty side-effect which result that Export modify hosts element, which looks wrong from my POV.

Comment (mchf) [inline src/modules/Host.rb]: yes, it is ugly, but it was so even before. See:
```
@hosts = Builtins.filter(@hosts) { |_ip, names| names != [] }
```

Comment (jreidinger) [inline src/modules/Host.rb]: hmm, so please add at least fixme to remove such side effect.

**Comment by mvidner** [comment]
LGTM.

File: src/modules/Host.rb [hunk 4]
```diff
@@ -207,14 +207,15 @@ def Import(settings)
...
-        return { "hosts" => Builtins.eval(@hosts) }
-      else
-        return {}
-      end
+      return {} if hosts.empty?
+
+      # Filter out IPs with empty hostname (so that valid autoyast
+      # profile is created)(#335120)
+      # FIXME: this also removes records with empty names from @hosts. Such
+      # side effect is unexpected and should be removed.
```
Comment (mvidner) [inline src/modules/Host.rb]: A link to the original bug, in which  alternative approaches are discussed: https://bugzilla.suse.com/show_bug.cgi?id=335120

Comment (mchf) [inline src/modules/Host.rb]: thanks for the reference

**Comment by mchf** [comment]
Rebased.
Thanks for reviews. I'm going to merge it once travis is done.

**Pull request merged**